x-pack/plugins/features/server/plugin.ts

```diff
@@ -102,4 +75,21 @@ export class Plugin {
   public stop() {
     this.logger.debug('Stopping plugin');
   }
+
+  private registerOssFeatures(savedObjects: SavedObjectsServiceStart) {
+    const registry = savedObjects.getTypeRegistry();
+    const savedObjectTypes = registry
+      .getAllTypes()
+      .filter(t => !t.hidden)
```

Comment:
@pgayvallet I wrongly assumed that `getAllTypes` should return already filtered types, as `savedObjects.type` did in LP. Do we consider it's a problem that a plugin can expose all types to a consumer?

Reply:
When we use the type registry in the repository we need to know all types, but I think consumers of this API should usually only operate on types that are visible so I think it would make sense to make that the default. If we have a use case where this should be exposed outside of core we could have an option like `includedHiddenTypes: true`

Reply:
I named it `getAllTypes` in an attempt to distinguish it from the legacy `types`, but it's probably not enough and jsdoc is lacking. 

https://github.com/elastic/kibana/blob/358d13919bc597d13e6a98ec97fae2615093027e/src/legacy/server/saved_objects/saved_objects_mixin.js#L81-L82

It shouldn't be a problem for a plugin to access / expose all types to a consumer. AFAIK, hidden types were already consumed in legacy, but for different usages, via the `schema` property that contains both visible and hidden types. Also I think some consumers needs access all types, even if I can't remember exactly for which usages, probably security/spaces)

However I agree that 1/ the jsdoc for `getAllTypes` should be more explicit, and 2/ the type registry should have a `getVisibleTypes` method.

I will create a issue for that improvement

Reply:
created https://github.com/elastic/kibana/issues/66818